service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/utils/EncounterExtractor.java

```diff
@@ -0,0 +1,54 @@
+package uk.nhs.adaptors.gp2gp.ehr.utils;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+import org.hl7.fhir.dstu3.model.Bundle;
+import org.hl7.fhir.dstu3.model.Encounter;
+import org.hl7.fhir.dstu3.model.ListResource;
+import org.hl7.fhir.dstu3.model.Reference;
+import org.hl7.fhir.dstu3.model.ResourceType;
+import org.springframework.stereotype.Component;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+@Component
+public final class EncounterExtractor {
+    private static final String CONSULTATION_LIST_CODE = "1149501000000101";
+
+    public static List<Encounter> extractConsultationListEncounters(List<Bundle.BundleEntryComponent> entries) {
```

Comment:
Are the encounters in the returned `List<Encounter>` in the same order at they appear in the Encounter List? I'd like to assert this is the case.

Reply:
Yes, their order is being asserted in a test. Order is set according to references order in the consultation list.

Reply:
The test works because the bundle happens to have the encounters in the same order as the encounter list.

I create a branch where I swapped the order of two encounters in the bundle to make the test fail. https://github.com/nhsconnect/integration-adaptor-gp2gp/pull/87/files

We can go through the code tomorrow to look at why if you want.